internal/api/courses/users/drop_test.go

```diff
@@ -0,0 +1,111 @@
+package users
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/util"
+	"reflect"
+	"testing"
+)
+
+func TestUserDrop(test *testing.T) {
+	defer db.ResetForTesting()
+
+	course := db.MustGetTestCourse()
+
+	testCases := []struct {
+		email     string
+		target    string
+		permError bool
+		locator   string
+		expected  DropResponse
+	}{
+		// Valid permissions.
+		{"course-admin", "course-other@test.edulinq.org", false, "", DropResponse{true}},
+		{"course-admin", "course-student@test.edulinq.org", false, "", DropResponse{true}},
+		{"course-owner", "course-grader@test.edulinq.org", false, "", DropResponse{true}},
+		{"course-owner", "course-admin@test.edulinq.org", false, "", DropResponse{true}},
+
+		// Valid permissions, role escalation.
+		{"server-owner", "course-other@test.edulinq.org", false, "", DropResponse{true}},
+		{"server-owner", "course-student@test.edulinq.org", false, "", DropResponse{true}},
+		{"server-admin", "course-grader@test.edulinq.org", false, "", DropResponse{true}},
+		{"server-admin", "course-admin@test.edulinq.org", false, "", DropResponse{true}},
+		{"server-admin", "course-owner@test.edulinq.org", false, "", DropResponse{true}},
+
+		// Invalid permissions.
+		{"course-student", "course-other@test.edulinq.org", true, "-020", DropResponse{false}},
+		{"course-grader", "course-other@test.edulinq.org", true, "-020", DropResponse{false}},
+
+		// Invalid permissions, role escalation.
+		{"server-creator", "server-user@test.edulinq.org", true, "-040", DropResponse{false}},
+		{"server-user", "server-user@test.edulinq.org", true, "-040", DropResponse{false}},
+
+		// Target user is not found.
```

Comment:
Add test for valid permission, but the user us not enrolled in the course.
I'm fine with the API response being the same as it is now (no error or warning),
since by the end of the API call the user will not be enrolled in the course (even if they were not enrolled before).